test/integration/consul-container/test/gateways/http_route_test.go

```diff
@@ -186,73 +185,229 @@ func TestHTTPRouteFlattening(t *testing.T) {
 	libassert.CatalogServiceExists(t, client, gatewayName)
 
 	//make sure config entries have been properly created
-	require.Eventually(t, func() bool {
-		entry, _, err := client.ConfigEntries().Get(api.APIGateway, gatewayName, &api.QueryOptions{Namespace: namespace})
-		assert.NoError(t, err)
-		if entry == nil {
-			return false
-		}
-		apiEntry := entry.(*api.APIGatewayConfigEntry)
-		t.Log(entry)
-		return isAccepted(apiEntry.Status.Conditions)
-	}, time.Second*10, time.Second*1)
-
-	require.Eventually(t, func() bool {
-		entry, _, err := client.ConfigEntries().Get(api.HTTPRoute, routeOneName, &api.QueryOptions{Namespace: namespace})
-		assert.NoError(t, err)
-		if entry == nil {
-			return false
-		}
-
-		apiEntry := entry.(*api.HTTPRouteConfigEntry)
-		t.Log(entry)
-		return isBound(apiEntry.Status.Conditions)
-	}, time.Second*10, time.Second*1)
-
-	require.Eventually(t, func() bool {
-		entry, _, err := client.ConfigEntries().Get(api.HTTPRoute, routeTwoName, nil)
-		assert.NoError(t, err)
-		if entry == nil {
-			return false
-		}
-
-		apiEntry := entry.(*api.HTTPRouteConfigEntry)
-		return isBound(apiEntry.Status.Conditions)
-	}, time.Second*10, time.Second*1)
+	checkGatewayConfigEntry(t, client, gatewayName, namespace)
+	checkHTTPRouteConfigEntry(t, client, routeOneName, namespace)
+	checkHTTPRouteConfigEntry(t, client, routeTwoName, namespace)
 
 	//gateway resolves routes
-	ip := "localhost"
 	gatewayPort, err := gatewayService.GetPort(listenerPort)
 	assert.NoError(t, err)
 
+	//route 2 with headers
+
 	//Same v2 path with and without header
-	checkRoute(t, ip, gatewayPort, "v2", map[string]string{
+	checkRoute(t, gatewayPort, "v2", map[string]string{
```

Comment:
#nit I might continue to use `"/v2"` in these scenarios because it hints at what the variable is when you're reading through the code without knowing the internals of `checkRoute`